Reference documentation for the CSS Custom highlight API (#23907)
mdn/content: files/en-us/web/api/highlightregistry/keys/index.md

---
title: HighlightRegistry.keys()
slug: Web/API/HighlightRegistry/keys
page-type: web-api-instance-method
tags:
  - API
  - Method
  - CSS Custom Highlight API
  - Reference
  - keys
browser-compat: api.HighlightRegistry.keys
---

{{APIRef("CSS Custom Highlight API")}}

> The **`keys()`** method of the {{domxref("HighlightRegistry")}} interface returns a new [Iterator](/en-US/docs/Web/JavaScript/Guide/Iterators_and_Generators) object that
> contains the keys for each `Highlight` object in the `HighlightRegistry` object in insertion order.

Reviewer: ```suggestion
The **`keys()`** method of the {{domxref("HighlightRegistry")}} interface returns a new [Iterator](/en-US/docs/Web/JavaScript/Guide/Iterators_and_Generators) object that contains the names of each `Highlight` object in the `HighlightRegistry` object in insertion order.
```

`HighlightRegistry` is a {{jsxref("Map")}}-like object, so this is similar to using {{jsxref("Map.keys()")}}.

## Syntax

```js-nolint
keys()
```

### Return value

> A new iterator object containing the keys for each `Highlight` object in the registry, in insertion order.

Reviewer: ```suggestion
A new iterator object containing the names of each `Highlight` object in the registry, in insertion order.
```
